python/lsst/ts/MTAlignment/alignment_model.py

```diff
@@ -261,14 +265,14 @@ async def _basic_send_command(self, cmd: str) -> str:
             raise RuntimeError(err_msg)
 
         reply_str = reply_bytes.decode().strip()
-        if reply_str == "EMP":
-            return reply_str
         reply_match = self.reply_regex.match(reply_str)
         if reply_match is None:
             raise RuntimeError(f"Cannot parse reply {reply_str!r}")
         reply_type, reply_code, reply_body = reply_match.groups()
         if reply_type == "ACK":
             return reply_body
+        elif reply_type == "ERR" and reply_code == BUSY_ERR_CODE_STR:
```

Comment:
Other errors are silently ignored. Is that intentional? If yes, please add a comment here explaining that. If no then please add some basic code that handles other error codes as well.

Reply:
Other errors raise an exception 2 lines later. But you raise an excellent point. This code would be much easier to understand and less awkward if all ERR-xxx replies were handled the same.

I have reworked the code to do that, eliminating the BUSY_REPLY constant. The new code has the following changes in `AlignmentModel`:
* `get_status` (formerly `check_status`) catches the busy exception (ERR-201) and returns "BUSY". This reflects how `get_status` is used: a caller will want to know if the system is busy without having to do fancy exception handling.
* `wait_for_ready` now calls `get_status` instead of using `_basic_send_command` to send the status command. The only reason it didn't already do that, is that `get_status` locked the communication lock, which `wait_for_ready` must not do. but now that `get_status` has this extra exception handling code, which I really don't want to duplicate, I decided to add an optional `dolock` argument to `get_status`.

I think the result is much cleaner. The only down side is that a future maintainer can call `get_status` with `dolock=False` when it should be true. But that seems unlikely and it will usually raise an exception (it won't if the lock is already obtained, but that should be rare).